### PR TITLE
Connect CDN to correct S3 bucket & restrict access by using an OAI

### PR DESCRIPTION
**Pull request by mablanco**
Please read the discussion at #1474.

### CODE REVIEW

**Comment by mablanco** [comment]
These latest commits solve the following issues:

- Blocked all public access to S3 bucket (we're using OAI now).
- Atomatically update bucket policy to grant access to the OAI.
- Remove the `origin_path` attributes from both origins; combined with the `path_pattern` attribute (necessary for choosing origin per request) makes Cloudfront expect nested directories in the origins.
- Make the S3 origin the default one and removed its `path_pattern` attribute. Now you can save your static website wherever you want inside the bucket.
- Fix the value of the `path_pattern` attribute for the API GW origin.
- Connect APIGW origin to correct URL instead of ELB (#1476).
- Improve readability by reordering attributes and blocks.

File: infra/aws/terraform/cdn.tf [hunk 1]
```diff
@@ -38,48 +38,52 @@ resource "aws_cloudfront_distribution" "this" {
...
-
   restrictions {
     geo_restriction {
       restriction_type = "none"
     }
   }
+
+  default_cache_behavior {
+    allowed_methods        = ["GET", "HEAD"]
+    cache_policy_id        = "658327ea-f89d-4fab-a63d-7e88639e58f6"
```
Comment (zimny) [inline infra/aws/terraform/cdn.tf:49]: I suppose this is ID of some managed policy, can you use the datasource instead?
eg
```
data "aws_cloudfront_cache_policy" "this" {
  name = "Managed-CachingOptimized"
}
...
 cache_policy_id          = data.aws_cloudfront_cache_policy.this.id
```
?

Comment (mablanco) [reply infra/aws/terraform/cdn.tf:49]: I'd rather use the `id` as it's not subject to changes as a name can be. However, I understand it's not easy to read and it makes you go to the AWS docs to know about that policy or at least add a comment to state which policy you're using.

What path do you prefer?

Comment (zimny) [reply infra/aws/terraform/cdn.tf:49]: well, magic string is in most cases less preferable than data source :) if possible please use the data source instead

Comment (mablanco) [reply infra/aws/terraform/cdn.tf:49]: ok, I'll go that way

**Review by zimny — APPROVED** [review]
LGTM

**Pull request merged**